src/main/java/racingcar/RacingGame.java

```diff
@@ -1,32 +1,58 @@
 package racingcar;
 
-import racingcar.view.ResultView;
-
 import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 public class RacingGame {
+
+    private static final int MINIMUM_CAR_COUNT = 2;
+    private static final int MINIMUM_ROUND = 1;
+
     private int round;
-    private Car[] cars;
+    private Cars cars;
 
-    public RacingGame(int round, Car[] cars){
+    public RacingGame(int round, String[] carNames) {
+        validate(round, carNames);
         this.round = round;
-        this.cars = cars;
+        this.cars = generateCars(carNames);
     }
 
-    public int[] play() {
+    public Cars getCars() {
+        return cars;
+    }
+
+    private Cars generateCars(String[] carNames) {
+        List<Car> carList = Arrays.stream(carNames)
+                .map(CarName::of)
+                .map(Car::newInstance)
+                .collect(Collectors.toList());
+
+        return new Cars(carList);
+    }
+
+    public RacingGameResult play() {
+
+        cars.clearAll();
+
+        RacingGameResult racingGameResult = new RacingGameResult();
         IntStream.range(0, round)
-                .mapToObj(value -> moveAll())
-                .forEach(ResultView::printCarPositionPerRound);
-
-        return Arrays.stream(cars)
-                .mapToInt(Car::getPosition)
-                .toArray();
-    };
-
-    private int[] moveAll() {
-      return Arrays.stream(cars)
-                .mapToInt(Car::move)
-                .toArray();
+                .forEach(value -> {
+                    cars.moveAll();
+                    racingGameResult.addFromCars(cars);
+                });
+
+        return racingGameResult;
+    }
+
+    private void validate(int gameRound, String[] carNames) {
+        if (Objects.isNull(carNames) || carNames.length < MINIMUM_CAR_COUNT) {
+            throw new IllegalArgumentException("최소 자동차 대수 입력 값은 " + MINIMUM_CAR_COUNT + " 입니다.");
+        }
+        if (gameRound < MINIMUM_ROUND) {
+            throw new IllegalArgumentException("최소 게임 라운드 입력 값은 " + MINIMUM_ROUND + " 입니다.");
+        }
```

Comment:
개인적인 생각으론 두 검증 과정 순서가 매개변수 순서에 맞게 변경되면 좋을 것 같아요:)
```suggestion
        if (gameRound < MINIMUM_ROUND) {
            throw new IllegalArgumentException("최소 게임 라운드 입력 값은 " + MINIMUM_ROUND + " 입니다.");
        }
        if (Objects.isNull(carNames) || carNames.length < MINIMUM_CAR_COUNT) {
            throw new IllegalArgumentException("최소 자동차 대수 입력 값은 " + MINIMUM_CAR_COUNT + " 입니다.");
        }
```